script/reviewer.go

```diff
@@ -19,11 +20,84 @@ func (r *Reviewer) Review() {
 
 	r.application.Parse(r.gitHub.Issue)
 
-	if isTestingIssue() {
-		if r.application.IsValid() {
-			debugMessage("Application has no problems")
-		} else {
-			debugMessage("Application problems:", r.application.RenderProblems())
+	status := r.getStatus()
+	r.updateLabels(status)
+	r.createComment(status)
+}
+
+func (r *Reviewer) getStatus() string {
+	if *r.gitHub.Issue.State == "closed" {
+		return "closed"
+	} else if r.gitHub.IssueHasLabel(LabelStatusApproved) {
+		return "approved"
+	} else if r.gitHub.IssueHasLabel(LabelStatusReviewing) {
+		return "reviewing"
+	} else if r.gitHub.IssueHasLabel(LabelStatusInvalid) {
+		return "invalid"
+	} else {
+		return "new"
+	}
+}
+
+func (r *Reviewer) createComment(status string) {
+	title := ""
+	body := ""
+	details := fmt.Sprintf("<details>\n<summary>Application data...</summary>\n\n```json\n%s\n```\n</details>", r.application.GetData())
+
+	if status == "closed" {
+		body = "Oops! This application is closed can no longer be processed. If this is an error, please reach out to [opensource@1password.com](mailto:opensource@1password.com)."
+	} else if status == "approved" {
+		body = "Oops! This application has been updated but has already been approved and can no longer be processed. If this is an error, please reach out to [opensource@1password.com](mailto:opensource@1password.com)."
```

Comment:
> This application has been updated but has already been approved and can no longer be processed. If this is an error, please reach out to

This sentence is not clear enough about what the user did wrong and actions the user needs to take next. What is the _this_ in _If this is an error_?

Do you mean something like:

> We have already processed and approved this application. If you’d like to make an edit to it, please email [opensource@1password.com](mailto:opensource@1password.com).

Reply:
In this scenario the application is approved and the details of their program participation are already committed to the repository. So this actually makes me think I can even more helpful here:

> This application has already been approved and changes will not be reviewed. If you would like to modify the details of your application, submit a PR against the stored [application data](%s). If this is an error, contact us at [opensource@1password.com](mailto:opensource@1password.com).

Reply:
I love that improvement except for _If this is an error_. If I'm being too pedantic here, feel free to ignore me. The _this_ in that clause is still nebulous. If you have something specific in mind, dereference that pronoun[^1] and put it in there. I think that you're trying for a catch-all instruction. In which case, maybe something more like, _If you have any questions,_, which doesn't indicate that there is something necessarily wrong.

[^1]: Pronouns are just pointers for English, if you think about it.

Reply:
Yep, looking for that "if for any reason this message is appearing at an unexpected time" catch-all response. I will use your latter suggestion!